bindings/pyroot/src/Utility.cxx

```diff
@@ -254,6 +254,26 @@ Bool_t PyROOT::Utility::AddToClass( PyObject* pyclass, const char* label, PyCall
    return kTRUE;
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// Add the given function with name 'func' to the class as read-only property 'property'.
+
+Bool_t PyROOT::Utility::AddProperty(PyObject *pyclass, const char *func, const char *property)
+{
+   PyObject *builtin = PyImport_ImportModule("__builtin__");
+   PyObject *property_class = PyObject_GetAttrString(builtin, "property");
+   PyObject *func_obj = PyObject_GetAttrString(pyclass, func);
+   PyObject *arglist = Py_BuildValue("(O)", func_obj);
+   PyObject *property_obj = PyObject_CallObject(property_class, arglist);
+   PyObject_SetAttrString(pyclass, property, property_obj);
+
+   Py_DECREF(builtin);
+   Py_DECREF(property_class);
+   Py_DECREF(func_obj);
+   Py_DECREF(arglist);
+
```

Comment:
I wonder whether we need to invoke `Py_DECREF` on `property_obj` here, like it is done with `method` in `AddToClass`?

Reply:
Now I wonder the same. The test still run with an additional `Py_DECREF` on the `property_obj` but from my understanding we should keep a reference count of 1 for the actual property object. Probably `PyDict_SetAttrString` does increase the reference count? I'll need to look this up.